MediaBrowser.Controller/Tasks/IBaseItemScheduledTask.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Model.Tasks;
+
+namespace MediaBrowser.Controller;
+
+/// <summary>
+/// A scheduled task that can be run in the context of one or many base items.
+/// </summary>
+public interface IBaseItemScheduledTask : IScheduledTask
+{
+    /// <summary>
+    /// Runs this task in the context of one or many base items.
+    /// </summary>
+    /// <param name="progress">The progress.</param>
+    /// <param name="baseItems">The list of items that where changed.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task.</returns>
+    Task ExecuteAsync(IProgress<double> progress, IReadOnlyList<(BaseItem Item, TaskActionTypes ChangeType)> baseItems, CancellationToken cancellationToken);
```

Comment:
Might be more efficient to send a list of baseitems instead of calling the execute function multiple times.

Reply:
Please elaborate, there is already a list of baseItems with thier associated action as a parameter.

Reply:
Whoops, I did't read the list part of the signature. nvm my comment